### ISSUE TITLE
`object` methods works only for hashes

### ISSUE THREAD

**Issue by jomei** [inre/model_pack#2]
Hello

I'm trying to use `object` method to wrap up some not hash data to my custom class and it doesn't work.

For instance, I have a class
```ruby
class SomeClass
   object :foo, class_name: Foo
   object :bar, class_name: Bar
   object :baz, class_name: Baz
end
```
And some hash like 
```
my_data =  {
   foo: {some: 'Hash'},
   bar: 'some_string'
   baz: 42
}
```
Trying to create `SomeClass` instance with this data
```ruby
my_instance = SomeClass.new(my_data)
```
Expected result:
`my_instance` has foo, bar, baz field with Foo, Bar, Baz instances inside accordingly.

Actual result:
`my_instance` has foo, bar, baz field with Foo instance, 'some_string' and 42.

Any suggestions?

**Comment by inre** [comment]
I don't think that is good idea. Because the constructor only apply named parameters so it can make extra complexity in there.

Wrong,

```ruby
class Bar
  attribute :str
  def initialize(str)
    @str = str
  end
end
```

Good,

```ruby
class Bar
attribute :str
end
a = Bar.new(str: "str")
```

for String just use,

```ruby
class SomeClass
   attribute :str
end
```

How do you plan to extend the class with fixed arguments?

**Comment by jomei** [comment]
> for String just use

Unfortunately, it doesn't cover my case, because I can't pass `class_name` into  `attribute`  method. 
Please check out specs in linked PR for better understanding of my problem.
